data/crac-io/crac-io-json/src/main/java/com/powsybl/openrao/data/craciojson/deserializers/NetworkActionArrayDeserializer.java

```diff
@@ -42,37 +42,50 @@ public static void deserialize(JsonParser jsonParser, String version, Crac crac,
                     case OPERATOR:
                         networkActionAdder.withOperator(jsonParser.nextTextValue());
                         break;
+                    case TRIGGER_CONDITIONS:
+                        jsonParser.nextToken();
+                        TriggerConditionDeserializer.deserialize(jsonParser, networkActionAdder);
+                        break;
                     case ON_INSTANT_USAGE_RULES:
+                        checkVersionForUsageRules(version);
                         jsonParser.nextToken();
                         OnInstantArrayDeserializer.deserialize(jsonParser, networkActionAdder);
```

Comment:
what happens if you use the new TriggerConditionDeserializer.deserialize(jsonParser, networkActionAdder) here ? I think it might work (given some small adjustments), and it would allow you to remove all the old deserializers